### PR TITLE
Arithmetic operations should not overflow

### PR DESCRIPTION
**Pull request by pontusmelke**
Before this fix `RETURN 9223372036854775808` would fail but `RETURN 9223372036854775807 + 1'
would overflow, now both will fail with nice error messages.

### CODE REVIEW

File: community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/helpers/TypeSafeMathSupport.scala [hunk 1]
```diff
@@ -121,102 +128,112 @@ trait TypeSafeMathSupport {
...
-      case (l: Short, r: Short)  => l * r
+    try {
+      (left, right) match {
+        case (null, _) => null
+        case (_, null) => null
+
+        case (l: Byte, r: Byte) => l * r
+        case (l: Byte, r: Double) => l * r
+        case (l: Byte, r: Float) => l * r
+        case (l: Byte, r: Int) => Math.multiplyExact(l, r.toLong)
```
Comment (Mats-SX) [inline community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/helpers/TypeSafeMathSupport.scala:194]: Wouldn't this always fit in a long? 2^8 \* 2^32 = 2^40 < 2^64? And the same for some of the below operations, that doesn't include a `long`, or am I missing something?

File: community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/helpers/StringRenderingSupport.scala [hunk 2]
```diff
@@ -1,37 +0,0 @@
...
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-package org.neo4j.cypher.internal.helpers
-
-/**
- * Subclasses prefer to use StringBuilders to render themselves
- */
-// TODO: This is only here for accomodating cypher-compiler-1.9. Do not touch, do not import, purge it with fire post 2.3
```
Comment (Mats-SX) [inline community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/helpers/StringRenderingSupport.scala:25]: Woohoo!

File: community/cypher/cypher/src/test/scala/org/neo4j/cypher/SemanticErrorAcceptanceTest.scala [hunk 3]
```diff
@@ -390,6 +390,51 @@ class SemanticErrorAcceptanceTest extends ExecutionEngineFunSuite {
     )
   }
 
+  test("should warn when addition overflows") {
```
Comment (Mats-SX) [inline community/cypher/cypher/src/test/scala/org/neo4j/cypher/SemanticErrorAcceptanceTest.scala:393]: Technically it's an error.

Comment (pontusmelke) [inline community/cypher/cypher/src/test/scala/org/neo4j/cypher/SemanticErrorAcceptanceTest.scala:393]: I ~~copy pasted~~ adopted the naming convention from the other tests :)

**Comment by jexp** [comment]
@pontusmelke was there a reason why we only fixed that in 3.0 ?

**Pull request merged**